loghero/LogHeroPluginSettings.php

```diff
@@ -2,19 +2,139 @@
 
 namespace LogHero\Wordpress;
 use \LogHero\Client\LogTransportType;
+use LogHero\Client\RedisOptions;
 
 
 class LogHeroPluginSettings {
```

Comment:
It's a bit hard to understand what's going. Especially with the two static calls where a new instance is created. This class seems to have multiple purposes like reading and writing the settings, "heavy" initialization and being a container for the settings. I would recommend refactoring this class and the way the settings get loaded, saved and initialized. For clearer responsibility and a more readable and understandable overall flow.

Reply:
I know that and I am not proud of that construct :). However, it really helps to have all the settings stuff in one place and switch depending on the requirements. This class does not actually save the settings (this is done in the admin view) but it synchronizes the DB settings with the settings stored in the JSON file. I need the settings from the JSON file for async flush because I am not allowed to load any wordpress ressources there. This is what makes reading the settings so complicated.